test/integration/multiple_deposit.test.ts

```diff
@@ -125,169 +143,648 @@ contract<ERC20RootVault, DeployOptions, CustomContext>(
+                        this.erc20Vault.address
+                    );
+                    const wethAmountOnYearnVault = await this.weth.balanceOf(
+                        this.yearnVault.address
+                    );
+                    const usdcAmountOnYearnVault = await this.usdc.balanceOf(
+                        this.yearnVault.address
+                    );
+                    const additionalUsdcAmount = usdcAmountOnERC20Vault.add(
+                        usdcAmountOnYearnVault
+                    );
+                    const additionalWethAmount = wethAmountOnERC20Vault.add(
+                        wethAmountOnYearnVault
+                    );
+                    if (additionalUsdcAmount.gt(0)) {
+                        await mint(
+                            "USDC",
+                            this.deployer.address,
+                            additionalUsdcAmount
+                        );
+                    }
+                    if (additionalWethAmount.gt(0)) {
+                        await mint(
+                            "WETH",
+                            this.deployer.address,
+                            additionalWethAmount
+                        );
+                    }
+                    await this.weth
+                        .connect(this.deployer)
+                        .transfer(
+                            this.erc20Vault.address,
+                            wethAmountOnERC20Vault
+                        );
+                    await this.usdc
+                        .connect(this.deployer)
+                        .transfer(
+                            this.erc20Vault.address,
+                            usdcAmountOnERC20Vault
+                        );
+
+                    await this.weth
+                        .connect(this.deployer)
+                        .transfer(
+                            this.yearnVault.address,
+                            wethAmountOnYearnVault
+                        );
+                    await this.usdc
+                        .connect(this.deployer)
+                        .transfer(
+                            this.yearnVault.address,
+                            usdcAmountOnYearnVault
+                        );
+
+                    let pricesResult = await this.mellowOracle.price(
+                        this.usdc.address,
+                        this.weth.address,
+                        0x28
+                    );
+                    let pricesX96 = pricesResult.pricesX96;
+                    let averagePrice = BigNumber.from(0);
+                    for (let i = 0; i < pricesX96.length; ++i) {
+                        averagePrice = averagePrice.add(pricesX96[i]);
+                    }
+                    averagePrice = averagePrice.div(pricesX96.length);
+
+                    let tvls = await this.subject.tvl();
+                    let minTvl = tvls[0];
+
+                    let withdrawAmounts: BigNumber[] = [];
```

Comment:
This all needs to be broken down into smaller pieces. It's impossible to read 500 lines of code with no sections / functions breakdown